src/MainDriver.cpp

```diff
@@ -850,11 +872,19 @@ int main(Global& glb, const char* souffle_executable) {
 
     // parse file
     ErrorReport errReport(process_warn_opts(glb));
-
     DebugReport debugReport(glb);
-    Own<ast::TranslationUnit> astTranslationUnit = ParserDriver::parseTranslationUnit(
-            glb, InputPath.string(), Input->getInputStream(), errReport, debugReport);
-    Input->endInput();
+
+    std::string sourceBuffer;
+    Input->getInputStream();
+    if (!Input->readInput(sourceBuffer)) {
```

Comment:
The calls to `getInputStream()` and `endInput()` should not be necessary if they are executed within `readInput()`.